src/ray/raylet/local_object_manager.cc

```diff
@@ -57,9 +60,16 @@ void LocalObjectManager::WaitForObjectFree(const rpc::Address &owner_address,
 }
 
 void LocalObjectManager::ReleaseFreedObject(const ObjectID &object_id) {
-  {
+  if (object_pinning_enabled_) {
```

Comment:
Why would it ever be false here?

Reply:
There are several tests that have this value False. It is also False when `--lru-evict` flag is one. 

We didn't need this before because we don't have the invariant check.

```cpp
    RAY_CHECK((pinned_objects_.count(object_id) > 0) ||
              (spilled_objects_url_.count(object_id) > 0) ||
              (objects_pending_spill_.count(object_id) > 0));
```

I prefer to have this check here in order to ensure strong consistency of internal data structures. (but lmk if you think that check is unnecessary and would prefer to remove this if statement).

I will add a comment here as well.